tests/js/client/shell/shell-index.js

```diff
@@ -0,0 +1,116 @@
+/*global fail, assertEqual, assertNotEqual, assertTrue, assertFalse, assertNull */
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test index methods
+///
+/// DISCLAIMER
+///
+/// Copyright 2010-2012 triagens GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is triAGENS GmbH, Cologne, Germany
+///
+/// @author Dr. Frank Celler, Lucas Dohmen
+/// @author Copyright 2012, triAGENS GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+const jsunity = require("jsunity");
+const internal = require("internal");
+const arango = internal.arango;
+const errors = internal.errors;
+const print = internal.print;
+const { helper, versionHas } = require("@arangodb/test-helper");
+const platform = require('internal').platform;
+
+
+const cn = "UnitTestsCollection";
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test suite: basics
+////////////////////////////////////////////////////////////////////////////////
+
+function IndexSuite() {
+  'use strict';
+
+  let collection = null;
+
+  return {
+
+    setUp : function () {
+      internal.db._drop(cn);
+      collection = internal.db._create(cn);
+    },
+
+    tearDown : function () {
+      internal.db._drop(cn);
+      collection = null;
+    },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test: indexes
+////////////////////////////////////////////////////////////////////////////////
+
+    testIndexProgress : function () {
+      var c = internal.db._create("c0l", {numberOfShards: 3, replicationFactor: 2});
+      var docs = [];
+      const sleep = require('internal').sleep;
+      for (var j = 0; j < 16; ++j) {
+        docs=[];
+        for (var i = 0; i < 10240; ++i) {
+          docs.push({name : "name" + i});
+        }
+        c.insert(docs);
+      }
+      
+      internal.debugSetFailAt("fillIndex::pause");
+      var idx = { name:"progress", type: "persistent", fields:["name"], inBackground: true};
+      arango.GET(`/_api/index?collection=c0l`);
+
+      var job = arango.POST_RAW(`/_api/index?collection=c0l`, idx,
+                                {"x-arango-async": "store"}).headers["x-arango-async-id"];
+      let count = 0;
+      var progress = 0.0;
+      var idxs = [];
+      while (true) {
```

Comment:
This whole test will not fail if one day no progress is reported at all. in fact, if would not have failed without this feature. The reason is that no assertion ever in here requires that `progress` is actually set!
We agreed to do this differently: We said that we set the pause failure point and then wait in the test until at least for once we see `progress` with a value between 0 and 100 (exclusive). Then we would release the pause with the second failure point and let it finish. In the meantime we can still check and if we see `progress`, it must be at least as large as the previous time (check non-decreasing). We should also check that `isBuilding` is set if and only if `progress` is set.